ethereum/core/src/main/java/tech/pegasys/teku/core/signatures/LocalMessageSignerService.java

```diff
@@ -43,22 +43,26 @@ public LocalMessageSignerService(final BLSKeyPair keypair, final AsyncRunner asy
   }
 
   @Override
-  public SafeFuture<BLSSignature> signAggregationSlot(final Bytes signingRoot) {
+  public SafeFuture<BLSSignature> signAggregationSlot(
+      final Bytes signingRoot, final Map<String, Object> additionalProperties) {
```

Comment:
I'm not sure this is the right approach.  We shouldn't need to create a map of properties when we already have the required data in a suitable format.  I'd suggest that the `MessageSignerService` interface is removed and just move everything up to the `Signer` interface which has the required data and is where the built-in slashing protection hooks in.

May need to extract a helper class to compute the signing root for data rather than having to duplicate that logic but that's doable.

Reply:
👍

Reply:
The primary purpose for this map was to serialize it as json to external signing service. If we delete MessageSignerService, that would mean LocalSigner and ExternalSigner extends Signer, is that correct assumption?

Reply:
Yep.

Reply:
Well, they'd implement `Signer` since its an interface but same idea. :)

Reply:
MessageSignerService has been removed. Refactored signingRoot calculation in a util class. Signer is now implemented directly by LocalSigner and ExternalSigner. Do we still want to wrap externalSigner in SlashingProtectedSigner (in ValidatorLoader)?

Reply:
Yes definitely by default, but (possibly in a follow up PR) it would be good to have an option to disable the built-in slashing protection for external signers (it should not be possible to disable it for local validators).

Reply:
👍  Thank you, that make sense.